src/main/java/use_case/Snowflake.java

```diff
@@ -0,0 +1,95 @@
+package main.java.use_case;
+
+/**
+ * Revised from Twitter's snowflake algorithm for generating unique IDs
+ * Github: https://github.com/twitter-archive/snowflake
+ */
+public class Snowflake {
+    private long workerId;
+    private long datacenterId;
+    private long sequence;
+
+    private final long workerIdBits = 5L;  // 5 bits for worker id
+    private final long datacenterIdBits = 5L;  // 5 bits for data center id
+    private final long sequenceBits = 12L;  // 12 bits for serial number
+
+    // maximum values for the IDs
+    private final long maxWorkerId = ~(-1L << workerIdBits);
+    private final long maxDatacenterId = ~(-1L << datacenterIdBits);
+    private final long maxSequenceBits = ~(-1L << sequenceBits);
+
+    // bit shifting for each id
+    private final long workerIdShift = sequenceBits;
+    private final long dataCenterIdShift = sequenceBits + workerIdBits;
+    private final long timestampShift = sequenceBits + workerIdBits + datacenterIdBits;
+
+    private final long initialTimeStamp = 1288834974657L;
```

Comment:
kind of small but maybe change the variable name to denote that this is the initial unix timestamp in milliseconds since unix is typically denoted in seconds.

Reply:
Fixed